mono/mini/memory-access.c

```diff
@@ -166,4 +166,25 @@ mini_emit_memory_load (MonoCompile *cfg, MonoClass *klass, MonoInst *src, int in
 	return ins;
 }
 
+
+void
+mini_emit_memory_store (MonoCompile *cfg, MonoClass *klass, MonoInst *dest, MonoInst *value, int ins_flag)
+{
+	MonoInst *ins;
+
+	if (ins_flag & MONO_INST_VOLATILE) {
+		/* Volatile stores have release semantics, see 12.6.7 in Ecma 335 */
+		mini_emit_memory_barrier (cfg, MONO_MEMORY_BARRIER_REL);
```

Comment:
For reference, here’s what CoreCLR does: https://github.com/dotnet/coreclr/blob/1969aa224e70a697d1622f7830df7b894b703189/src/inc/volatile.h

Does this have the same semantics?